include/dlaf/permutations/general.h

```diff
@@ -0,0 +1,47 @@
+//
+// Distributed Linear Algebra with Future (DLAF)
+//
+// Copyright (c) 2018-2022, ETH Zurich
+// All rights reserved.
+//
+// Please, refer to the LICENSE file in the root directory.
+// SPDX-License-Identifier: BSD-3-Clause
+//
+
+#pragma once
+
+#include <blas.hh>
+
+#include "dlaf/common/assert.h"
+#include "dlaf/matrix/matrix.h"
+#include "dlaf/permutations/general/api.h"
+#include "dlaf/util_matrix.h"
+
+namespace dlaf::permutations {
+
+template <Backend B, Device D, class T, Coord coord>
+void permutate(SizeType i_begin, SizeType i_end, Matrix<const SizeType, D>& perms, Matrix<T, D>& mat_in,
```

Comment:
I'm not sure which is better tbh... https://wikidiff.com/permutate/permute

Reply:
I would go for permute because:
1. It's shorter
2. At least according to https://en.wiktionary.org/wiki/permutate "Permute is much more commonly used".